### PR TITLE
Np 3213 update controlled vocabulary handler

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: customer/src/main/java/no/unit/nva/customer/ControlledVocabularyHandler.java [hunk 1]
```diff
@@ -24,4 +27,11 @@ public ControlledVocabularyHandler(Class<I> iclass, CustomerService customerServ
     protected final List<MediaType> listSupportedMediaTypes() {
         return SUPPORTED_MEDIA_TYPES;
     }
+
+
+    protected static UUID extractIdentifier(RequestInfo requestInfo) {
```
Comment (axthosarouris) [inline customer/src/main/java/no/unit/nva/customer/ControlledVocabularyHandler.java:32]: common method moved up to superclass

File: customer/src/main/java/no/unit/nva/customer/WriteControlledVocabularyHandler.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import com.amazonaws.services.lambda.runtime.Context;
+import java.util.UUID;
+import no.unit.nva.customer.model.CustomerDto;
+import no.unit.nva.customer.model.interfaces.VocabularySettingsList;
+import no.unit.nva.customer.service.CustomerService;
+import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+
+public abstract class WriteControlledVocabularyHandler
```
Comment (axthosarouris) [inline customer/src/main/java/no/unit/nva/customer/WriteControlledVocabularyHandler.java:11]: Contains the common logic of CreateVocabularySettings and UpdateVocabularySettings

File: customer/src/test/java/no/unit/nva/customer/create/CreateControlledVocabularyTest.java [hunk 3]
```diff
@@ -1,57 +1,31 @@
...
-import no.unit.nva.customer.testing.CustomerDynamoDBLocal;
-import no.unit.nva.testutils.HandlerRequestBuilder;
 import nva.commons.apigateway.GatewayResponse;
 import nva.commons.apigateway.MediaTypes;
 import nva.commons.apigateway.exceptions.ApiGatewayException;
-import nva.commons.core.Environment;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
-class CreateControlledVocabularyTest extends CustomerDynamoDBLocal {
```
Comment (axthosarouris) [inline customer/src/test/java/no/unit/nva/customer/create/CreateControlledVocabularyTest.java:37]: moved common setup logic for CreateContolledVocabularyTests and UpdateControlloedVocabularyTests  to common superclass `CreateUpdateControlledVocabularySettingsTests`

File: customer/src/test/java/no/unit/nva/customer/testing/CreateUpdateControlledVocabularySettingsTests.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import no.unit.nva.customer.service.impl.DynamoDBCustomerService;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import nva.commons.apigateway.GatewayResponse;
+import nva.commons.apigateway.MediaTypes;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.core.Environment;
+import nva.commons.core.JsonUtils;
+import org.junit.jupiter.api.BeforeEach;
+
+public abstract class CreateUpdateControlledVocabularySettingsTests extends CustomerDynamoDBLocal {
```
Comment (axthosarouris) [inline customer/src/test/java/no/unit/nva/customer/testing/CreateUpdateControlledVocabularySettingsTests.java:32]: Class creating common logic for tests regarding creating and updating vocabulary settings

**Review by brinxmat — APPROVED** [review]

**Pull request merged**